### DISCUSSION TITLE
Code search with Rest API, path qualifier

### DISCUSSION THREAD

**Discussion by sofiamigliorini** [community/community#64618] [Apps, API and Webhooks]
### Select Topic Area

General

### Body

Hi!
I'm trying to do a Python script in order to retrieve results for this search code query: 
"architecture.png" path:/(^|\/)README\.md$/
When I make my API call, I think there is a problem with the path qualifier, because if I omit it there is a result, otherwise there is no result for my query.

api_url = "https://api.github.com/search/code"

headers = {
    "Accept": "application/vnd.github+json",
    "Authorization": f"Bearer {ACCESS_TOKEN}",
    "X-Github-Api-Version": "2022-11-28",
}

params = {
    "q": "architecture.png",
    "per_page": "100"
}

response = requests.get(api_url, headers=headers, params=params)

that's my code! Is there someone who can help me please? :(

**Comment by ericyhwang** [comment]
Since I just ran into this myself...

The web code search experience was [updated in May 2023](https://github.blog/2023-05-08-github-code-search-is-generally-available/), with new search syntax.

As of now, the latest GitHub API version 2022-11-28 still uses a legacy code search syntax, documented here:
https://docs.github.com/en/search-github/searching-on-github/searching-code

One big difference is that the API doesn't support the new `path:` qualifier.

👉  As a workaround, using the old `filename:` qualifier in API queries can suffice in many cases.

It's reasonable that the API wouldn't switch to the new search syntax immediately, to avoid breaking existing integrations. However, it would be nice if the API docs were updated to note that the code search API still uses the older search syntax. I'd submit a PR to update them, but I can't find out where the [docs for the code search endpoint](https://docs.github.com/en/rest/search/search?apiVersion=2022-11-28#search-code) live.
